felix/fv/bpf_notracking_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fv_test
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+
+	"github.com/projectcalico/calico/felix/fv/connectivity"
+	"github.com/projectcalico/calico/felix/fv/containers"
+	"github.com/projectcalico/calico/felix/fv/infrastructure"
+	"github.com/projectcalico/calico/felix/fv/workload"
+	"github.com/projectcalico/calico/libcalico-go/lib/apiconfig"
+)
+
+var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ Felix bpf with NOTRACK feature", []apiconfig.DatastoreType{apiconfig.EtcdV3}, func(getInfra infrastructure.InfraFactory) {
+
+	if os.Getenv("FELIX_FV_ENABLE_BPF") != "true" {
+		// Non-BPF run.
+		return
+	}
+
+	const workloadTargetPort = "80"
+	var (
+		infra          infrastructure.DatastoreInfra
+		felixes        []*infrastructure.Felix
+		workloads      [2]*workload.Workload
+		cc             *connectivity.Checker
+		externalClient *containers.Container
+	)
+
+	BeforeEach(func() {
+		infra = getInfra()
+
+		opts := infrastructure.TopologyOptions{
+			FelixLogSeverity: "debug",
+			ExtraEnvVars: map[string]string{
+				"FELIX_BPFENABLED":              "true",
+				"FELIX_BPFHostConntrackBypass":  "true",
+				"FELIX_DEBUGDISABLELOGDROPPING": "true",
+			},
+		}
+		felixes, _ = infrastructure.StartNNodeTopology(2, opts, infra)
+		infra.AddDefaultAllow()
+		cc = &connectivity.Checker{}
+
+		for i := range felixes {
+			workloads[i] = workload.Run(felixes[i],
+				fmt.Sprintf("host%d-webserver", i),
+				"default",
+				fmt.Sprintf("10.65.%d.2", i),
+				workloadTargetPort, "tcp")
+			workloads[i].ConfigureInInfra(infra)
+		}
+
+		externalClient = infrastructure.RunExtClient("ext-client")
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			infra.DumpErrorData()
+		}
+
+		for _, felix := range felixes {
+			felix.Stop()
+		}
+
+		infra.Stop()
+	})
+
+	It("should allow 3rd party DNAT to workloads work", func() {
+		expectNormalConnectivity := func() {
```

Comment:
I am not sure whether this is needed. We expect that. There is no application listening on that port on the host. So it would be a total surprise if it worked. And we are not quite testing that DNAT in iptables work, we trust that iptables work and are tested :)

Reply:
I agree, but without this we cannot be sure if the actual test is passing by chance or because of the fix. Do you prefer to remove it?